clientapi/routing/directory.go

```diff
@@ -181,10 +181,37 @@ func SetLocalAlias(
 		return *resErr
 	}
 
+	maybeRoomID, err := spec.NewRoomID(r.RoomID)
```

Comment:
Yeah fairs, my thought with it was to dereference as soon as possible, so if there ever was a nil-dereference, it gets caught early - and because go doesn't let you shadow variables I needed a name for the variable which might be a nil. But since we're dereferencing it in the same function anyway it's probably fine, I'll swap it over 👍